docs/developer/architecture/core/uisettings-service.asciidoc

```diff
@@ -38,3 +92,66 @@ export class MyPlugin implements Plugin {
 }
 
 ----
+
+Registeration only requires a schema for validation on read and write, all {kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.uisettingsparams.md[other parameters] are optional.
+
+See:
+
+- {kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.uisettingsservicesetup.register.md[UI settings service Setup API docs]
+
+==== Migrations
+Because `uiSettings` are persisted in the `config` saved object, changing or removing a setting requires a migration of the whole saved object. For example, if we wanted to remove the `custom` setting in the example above and then also rename `my_setting:fourtyTwo` to `my_other_setting:fourtyTwo`, we'd have two migration entries, one for each change targeting the required versions:
+[source,typescript]
+----
+export const migrations = {
+  ...
+  '8.1.0': (doc: SavedObjectUnsanitizedDoc<any>): SavedObjectSanitizedDoc<any> => ({
+  ...doc,
+  ...(doc.attributes && {
+    attributes: Object.keys(doc.attributes).reduce(
+      (acc, key) =>
+        [ // other settings to remove for 8.1.0...
+          // owner: Team:Custom setting team owner
+          'custom',
+        ].includes(key)
+          ? {
+              ...acc,
+            }
+          : {
+              ...acc,
+              [key]: doc.attributes[key],
+            },
+      {}
+    ),
+  }),
+  references: doc.references || [],
+  }),
+  '8.2.0': (doc: SavedObjectUnsanitizedDoc<any>): SavedObjectSanitizedDoc<any> => ({
+    ...doc,
+    ...(doc.attributes && {
+      attributes: Object.keys(doc.attributes).reduce(
+        (acc, key) =>
+          key.startsWith('my_setting:')
+            ? {
+                ...acc,
+                [key.replace('my_setting', 'my_other_setting')]: doc.attributes[key],
+              }
+            : {
+                ...acc,
+                [key]: doc.attributes[key],
+              },
+        {}
+      ),
+    }),
+    references: doc.references || [],
+  }),
+  …
+}
+----
+{kib-repo}blob/{branch}/src/core/server/ui_settings/saved_objects/migrations.ts[`uiSettings` migrations] are declared directly in the `uiSettings` service.
+
+==== Security-related consideration
```

Comment:
I'd consider merging this with a dedicated section on advanced settings ui (see my example above)

Reply:
Done